src/main/java/org/jabref/cli/ArgumentProcessor.java

```diff
@@ -289,6 +294,45 @@ public void processArguments() {
         if (!cli.isBlank() && !loaded.isEmpty()) {
             uiCommands.add(new UiCommand.OpenDatabases(loaded));
         }
+
+        if (cli.isCheckConsistency()) {
+            checkConsistency();
+        }
+    }
+
+    private void checkConsistency() {
+        String fileName = cli.getCheckConsistency();
+        String outputFormat = cli.getCheckConsistencyOutputFormat().toUpperCase();
+
+        if (fileName == null) {
+            System.out.println(Localization.lang("No file specified for consistency check."));
+            return;
+        }
+
+        try {
+            Path filePath = Path.of(fileName);
+            ParserResult parserResult = OpenDatabase.loadDatabase(filePath,
+                    cliPreferences.getImportFormatPreferences(), fileUpdateMonitor);
+            BibDatabaseContext databaseContext = parserResult.getDatabaseContext();
+
+            BibliographyConsistencyCheck checker = new BibliographyConsistencyCheck();
+            BibliographyConsistencyCheck.Result result = checker.check(databaseContext.getDatabase().getEntries());
+
+            String outputFileName = fileName + "_consistency_check." + outputFormat.toLowerCase();
+            try (Writer writer = new FileWriter(outputFileName, StandardCharsets.UTF_8)) {
```

Comment:
Please rewrite. If `outputFormat.isEmpty()` then just output to `System.out` (and use TXT as format).